packages/kotti-input-number/src/InputNumber.vue

```diff
@@ -137,70 +148,91 @@ export default {
 	},
 	watch: {
 		value: {
-			handler(newNumber) {
+			handler(newNumber, oldNumber) {
 				const newString = toString(newNumber)
-				const shouldUpdate = this.currentValueNumber !== toNumber(newString)
+				const truncatedNumber = toNumber(newString)
+				const { min, max, step } = this
+
+				if (!isInRange({ max, min, offset: 0, value: truncatedNumber }))
+					throw new RangeError(
+						`KtInputNumber: encounted an out-of-range number "${newNumber}"`,
+					)
 
-				if (shouldUpdate) this.setValue(newString)
+				if (!isStepMultiple({ min, step, value: truncatedNumber }))
+					throw new Error(
+						`KtInputNumber: encounted a value "${newNumber}" that doesn't fit ((min + k * step): where k is an integer)`,
+					)
+
+				const shouldUpdate = oldNumber !== truncatedNumber
+				if (shouldUpdate) this.setInternalStringValue(newString)
 			},
 			immediate: true,
 		},
 	},
 	methods: {
 		decrementValue() {
 			if (!this.isDecrementEnabled) return
-			if (this.currentValueNumber === null) {
-				const defaultString = this.min === null ? '0' : this.min + ''
-				this.setValue(defaultString)
+			if (this.value === null) {
+				const defaultNumber = this.min || 0
+				this.emitInput(defaultNumber)
```

Comment:
now we only emit, and then the value will pass the new number (only source of truth is this.value)